### ISSUE TITLE
Doesn't work an array of an user type

### ISSUE THREAD

**Issue by Splarv** [citusdata/citus#2135]
```
On all nodes and coordinator
=> create type b as (x int,y int);
CREATE TYPE
On coordinator only 
=> create table a (a int, b b array);
CREATE TABLE
=> insert into a (a,b) values (0,array[(0,0)::b]);

First check pure PostgreSQL
INSERT 0 1
olleg@[local]:9700/electron
=> select * from a;
 a |     b     
---+-----------
 0 | {"(0,0)"}
(1 row)

All fine, now distributed
olleg@[local]:9700/electron
=> select create_distributed_table('a','a');
NOTICE:  Copying data from local table...
 create_distributed_table 
--------------------------
 
(1 row)

olleg@[local]:9700/electron
=> insert into a (a,b) values (1,array[(1,1)::b]);
INSERT 0 1
olleg@[local]:9700/electron
=> select * from a;
ERROR:  wrong element type
olleg@[local]:9700/electron
```
Ubuntu 17.10
$ dpkg -s postgresql-10|grep Version
Version: 10.1-1.pgdg17.04+1
$ dpkg -s postgresql-10-citus-7.3|grep Version
Version: 7.3.0.citus-1

**Comment by marcocitus** [comment]
The workaround is to set `citus.binary_master_copy_format` to off, though arguably this should be more automatic.

**Comment by marcocitus** [comment]
Fixed by #2798
